crates/bevy_render/src/lib.rs

```diff
@@ -187,13 +192,13 @@ impl Plugin for RenderPlugin {
                         bevy_utils::tracing::info_span!("stage", name = "reserve_and_flush")
                             .entered();
 
-                    // reserve all existing app entities for use in render_app
+                    // reserve the number of entities used in the previous frame for use in render_app
```

Comment:
The point of reserving the entities is so that the same entity from the main world can be used to refer to refer to an entity in the render world. If we don’t reserve the entire active main world entity address space, something could be spawned after what is reserved and then something else is supposed to refer to that. I’m not convinced this change is safe to do. I’ll need to think about it some more. Maybe @cart can beat me to it.

Reply:
Yup the approach in this PR breaks the intended behavior. This line isn't about reserving space for rendering entities. Its about ensuring that rendering entities don't get spawned in the "current app id space".

We should not merge this PR.

Reply:
When I made the change I was expecting it to not work at all so I could learn more about Bevy's internals. I'm curious why it doesn't obviously break things, though.